src/dune_rules/gen_rules.ml

```diff
@@ -457,49 +465,50 @@ let gen_melange_emit_rules sctx ~dir ({ stanza_dir; stanza } as for_melange) =
                (Path.Source.to_string parent_melange_emit_dir)
            ]))
 
-let empty_rules =
+let empty_rules ~dir =
   { Build_config.Rules.build_dir_only_sub_dirs =
-      Subdir_set.These (String.Set.of_keys automatic_sub_dirs_map)
+      Path.Build.Map.singleton dir
+        (Subdir_set.These (automatic_subdirs (Path.Build.explode dir)))
```

Comment:
Another duplicated code snippet makes me think that there is some module to be written where these helper functions are going to live.

Reply:
took care of this too in [522eaea](https://github.com/ocaml/dune/pull/7207/commits/522eaead7b8b3c01d3a67573fc0d4fb346a568be)